src/platform/ESP32/CHIPDevicePlatformConfig.h

```diff
@@ -48,39 +48,53 @@
 
 #endif /* CHIP_DEVICE_CONFIG_DEVICE_PRODUCT_ID */
 
+#if defined(CONFIG_ENABLE_MATTER_OVER_THREAD) && CONFIG_ENABLE_MATTER_OVER_THREAD
 #define CHIP_DEVICE_CONFIG_ENABLE_THREAD CONFIG_ENABLE_MATTER_OVER_THREAD
+#endif
+
 #define CHIP_DEVICE_CONFIG_ENABLE_THREAD_SRP_CLIENT CONFIG_OPENTHREAD_SRP_CLIENT
 #define CHIP_DEVICE_CONFIG_ENABLE_THREAD_DNS_CLIENT CONFIG_OPENTHREAD_DNS_CLIENT
 
-#if CONFIG_ENABLE_ETHERNET_TELEMETRY
+#if defined(CONFIG_ENABLE_ETHERNET_TELEMETRY) && CONFIG_ENABLE_ETHERNET_TELEMETRY
 #define CHIP_DEVICE_CONFIG_ENABLE_WIFI 0
 #define CHIP_DEVICE_CONFIG_ENABLE_ETHERNET 1
-#elif CONFIG_IDF_TARGET_ESP32H2
+#elif defined(CONFIG_IDF_TARGET_ESP32H2) && CONFIG_IDF_TARGET_ESP32H2
 #define CHIP_DEVICE_CONFIG_ENABLE_WIFI 0
 #else
 #define CHIP_DEVICE_CONFIG_WIFI_STATION_RECONNECT_INTERVAL CONFIG_WIFI_STATION_RECONNECT_INTERVAL
 #define CHIP_DEVICE_CONFIG_MAX_SCAN_NETWORKS_RESULTS CONFIG_MAX_SCAN_NETWORKS_RESULTS
 #define CHIP_DEVICE_CONFIG_WIFI_SCAN_COMPLETION_TIMEOUT CONFIG_WIFI_SCAN_COMPLETION_TIMEOUT
 #define CHIP_DEVICE_CONFIG_WIFI_CONNECTIVITY_TIMEOUT CONFIG_WIFI_CONNECTIVITY_TIMEOUT
-#if CHIP_DEVICE_CONFIG_ENABLE_WIFI_AP
+
+#if defined(CHIP_DEVICE_CONFIG_ENABLE_WIFI_AP) && CHIP_DEVICE_CONFIG_ENABLE_WIFI_AP
 #define CHIP_DEVICE_CONFIG_WIFI_AP_SSID_PREFIX CONFIG_WIFI_AP_SSID_PREFIX
 #define CHIP_DEVICE_CONFIG_WIFI_AP_CHANNEL CONFIG_WIFI_AP_CHANNEL
 #define CHIP_DEVICE_CONFIG_WIFI_AP_MAX_STATIONS CONFIG_WIFI_AP_MAX_STATIONS
 #define CHIP_DEVICE_CONFIG_WIFI_AP_BEACON_INTERVAL CONFIG_WIFI_AP_BEACON_INTERVAL
 #define CHIP_DEVICE_CONFIG_WIFI_AP_IDLE_TIMEOUT CONFIG_WIFI_AP_IDLE_TIMEOUT
 #endif /* CHIP_DEVICE_CONFIG_ENABLE_WIFI_AP */
+
 #define CHIP_DEVICE_CONFIG_ENABLE_WIFI_TELEMETRY CONFIG_ENABLE_WIFI_TELEMETRY
 #define CHIP_DEVICE_CONFIG_ENABLE_WIFI CHIP_DEVICE_CONFIG_ENABLE_WIFI_AP | CHIP_DEVICE_CONFIG_ENABLE_WIFI_STATION
-#endif // CONFIG_IDF_TARGET_ESP32H2
+#endif // CONFIG_ENABLE_ETHERNET_TELEMETRY
 
-#if CONFIG_ENABLE_ICD_SERVER
+#if defined(CONFIG_ENABLE_ICD_SERVER) && CONFIG_ENABLE_ICD_SERVER
 #define CHIP_DEVICE_CONFIG_ICD_SLOW_POLL_INTERVAL chip::System::Clock::Milliseconds32(CONFIG_ICD_SLOW_POLL_INTERVAL_MS)
 #define CHIP_DEVICE_CONFIG_ICD_FAST_POLL_INTERVAL chip::System::Clock::Milliseconds32(CONFIG_ICD_FAST_POLL_INTERVAL_MS)
-#endif // CHIP_CONFIG_ENABLE_ICD_SERVER
+#endif
 
+#if defined(CONFIG_ENABLE_CHIPOBLE) && CONFIG_ENABLE_CHIPOBLE
```

Comment:
If CONFIG_ENABLE_CHIPOBLE is defined and false, wouldn't you still want to set CHIP_DEVICE_CONFIG_ENABLE_CHIPOBLE false here?  This part does not make sense to me.  Similar for a bunch of the checks below....

In general, if these CONFIG_ values are just "on or off" and "on" always means "defined" and "off" means "not defined", we should be checking ifdef only, not the value, I would think.

Reply:
Fixed setting of `CHIP_*` configs and always setting them even if the configs from the ESP32 is undefined.

> In general, if these CONFIG_ values are just "on or off" and "on" always means "defined" and "off" means "not defined", we should be checking ifdef only, not the value, I would think.

You are right here, just performing an `#ifdef` check.